### ISSUE TITLE
Duplicate `get_hostname` implementations

### ISSUE THREAD

**Issue by lukesteensen** [vectordotdev/vector#14819]
We have [one implementation](https://github.com/vectordotdev/vector/blob/405b333869d0053cf9ee214196e8d76be38a82e8/src/lib.rs#L146-L148) of `get_hostname` for Vector itself, and a [separate one](https://github.com/vectordotdev/vector/blob/405b333869d0053cf9ee214196e8d76be38a82e8/lib/vrl/stdlib/src/get_hostname.rs#L3-L8) for the VRL stdlib. It's not a huge deal since they're both quite small and currently the same, but when we introduce changes like allowing hostname to be overridden by an env var we don't want them to get out of sync.

**Comment by tobz** [comment]
We should de-duplicate those, yes. One way to avoid circular dependencies would be to force VRL to take stubs to anything that performs "external" I/O, such as getting the hostname. We would simply configure it when Vector initializes the VRL runtime, or something to that effect.

This would allow deduplicating without circular dependencies, and could be a potential model to expand upon for exposing "external" I/O through controlled interfaces.

**Comment by bruceg** [comment]
> We should de-duplicate those, yes. One way to avoid circular dependencies would be to force VRL to take stubs

Or just move it to `vector_common`

**Comment by tobz** [comment]
Perhaps. That's definitely easier in the short term.

**Comment by lukesteensen** [comment]
Good point, `vector_common` is a reasonable short-term fix here. I still don't love that arrangement in general, but that's a different issue.

**Comment by JeanMertz** [comment]
Yeah, I'd be happy with moving it to vector_common for now, and we can provide an actual public API/trait in VRL in the future, when we need VRL to be completely separate from Vector.
